msteams-platform/concepts/messaging-extensions/create-extensions.md

```diff
@@ -339,22 +352,130 @@ This used to insert a card into the compose box as a result of a the command. It
               ]
             }
           ]
-        },
-        "preview": {
-          "contentType": "application/vnd.microsoft.card.thumbnail",
-          "content": {
-            "title": "85069: Create a cool app",
-            "images": [
+        }
+      }
+    ]
+  }
+}
+```
+
+### Respond with an adaptive card message sent from a bot
+
+> [!NOTE]
+> Responding with an adaptive card message sent from a bot is in [developer preview](~/resources/dev-preview/developer-preview-intro.md). Currently the only type of response attachment supported is an adaptive card.
+
+You can also respond to the submit action by inserting a message with an [adaptive card](~/concepts/cards/cards.md#adaptive-cards) into the channel with a bot. Your user will be able to preview the message before submitting it, and potentially edit/interact with it as well. This can be very useful in scenarios where you need to gather information from your users before creating an adaptive card response. The following scenario shows how you can use this flow to configure a poll without including the configuration steps in the channel message.
+
+1. The user clicks the messaging extension to trigger the task module.
+1. The user uses the task module to configure the poll.
+1. After submitting the configuration task module the app uses the information provided in the task module to craft an adaptive card and sends it as a `botMessagePreview` response to the client.
+1. The user can then preview the adaptive card message before the bot will inserts it into the channel. If the bot is not already a member of the channel, clicking `Send` will add the bot.
+1. Interacting with the adaptive card will change the message before sending it.
+1. Once the user clicks `Send` the bot will post the message to the channel.
+
+To enable this flow your task module should respond as in the example below, which will present the preview message to the user.
+
+```json
+{
+  "composeExtension": {
+    "type": "botMessagePreview",
+    "activityPreview": {
+      "attachments":  [
+        {
+          "contentType": "application/vnd.microsoft.card.adaptive",
+          "content": << Card Payload >>
+        }
+      ]
+    }
+  }
+}
+```
+
+Your message extension will now need to respond to two new types of interactions, `value.botMessagePreviewAction = "send"` and `value.botMessagePreviewAction = "edit"`. Below is an example of the `value` object you will need to process:
+
+```json
+{
+  "name": "composeExtension/submitAction",
+  "type": "invoke",
+  ...
+  "value": {
+    "botMessagePreviewAction": "send",
+    "botActivityPreview": [
+      {
+        "type": "message/card",
```

Comment:
This is good catch. It's a bug that I should've reverted SMBA types back to BotBuilder types. For now we can skip documenting the message type, because we're only using first card. @clearab for now we can let bot developer always grab the first attachment to send: 
```
if (request.botMessagePreviewAction === 'send') {
    let attachment = request.botActivityPreview[0].attachments[0];
    let msg = new builder.Message()
        .address(event.address)
        .addAttachment(attachment);
    teamChatConnector.send([msg.toMessage()], (e, newAddr) => { ... });
}
```

Reply:
@robin-liao Is this JSON payload example incorrect then? If the bot developer is meant to simply take the first element and send it blindly, how does that work when the message type is incorrect, and the attachment structure shown is different?

Also, why does `botActivityPreview` have `name` and `imdisplayname`?

Reply:
I think the conversation, name, and imdisplayname properties should be outside of the value object, and part of the composeExtension/submitAction?

Reply:
@robin-liao I still need to document the JSON payload, should that be type="message/card" or just "message"?